Adds feature to Edit Custom certificates.

rumansaleem opened the pull request:
Closes #1618 

To be Added soon...

itguy327 commented:
Is this now in an official release?

rumansaleem commented:
Hey @jc21 can you please review this PR?

asmwp commented:
I have tried to use this version of NPM and pulled the docker image. When running this in Docker on Ubuntu using Firefox 138.0.4 (64-bit) when I go to add an SSL and select 'Custom' I get javascript errors 'InternalError: too much recursion - in main.bundle.js related to this line isNew http://0.0.0.0:81/js/17.bundle.17.js?v=2.12.3:1'. Have tried clearing cache etc, no good.

rumansaleem commented:
hey @asmwp thanks for looking at this. I will look into this and get back with the fix.

rumansaleem commented:
I've fixed the issue, I thought `this.isNew()` inside a template function would refer to model function. Fixed by directly checking `id`. Create and Edit both features seems to work as expected now.

@asmwp can you have another look?

asmwp commented:
So I downloaded the latest using this

docker pull nginxproxymanager/nginx-proxy-manager-dev:pr-4425

I would have expected this to have given me the latest, 

it gives me this image ID

sha256:1540fe19eb930a9d1f914f9195f9a087fee7ed8a76ded2821e2b85f0ae6ae926 

but the digest ID is

sha256:3292946dddcc96ee34f638ceba18933251b2ccd8601db962e0d694862539c11c

Are the Digest ID and Image ID not supposed to be the same?

Anyway as it stands in Firefox the exact same issue persists but it works ok using Chrome in Ubuntu.

I have a test environment so I was able to delete absolutely everything and do a fresh setup.

jstack01 commented:
My knowledge is limited on how the code works. Is there a way that I can update the certificate files without using the web interface? I know I can simply replace the files in /data/custom_ssl/npm-#/ however, this does not update the expire date on the web interface. Long stories short, I would like to create a script that updates the cert files and updates the "Expires" date on the web interface without manual intervention.

Shown below is a screenshot of what I am referring to. The cert that is there has been renewed but it is showing as expired. 
<img width="873" height="195" alt="Screenshot 2025-10-07 114439" src="https://github.com/user-attachments/assets/050d61f6-cdee-483b-bf54-4018f446c2bc" />

rjblake commented:
@rumansaleem - do you know what checks are failing? Seems this commit cannot be built as per Jenkins. Would love to see this in the base release.

jc21 commented:
Looks good, this is commonly requested. The React branch has been merged and changed a lot of things. Backend only received a code refresh, nothing changed functionality wise.

Frontend is a different story though..

DaviidMM commented:
+1

rumansaleem commented:
Hey guys, Sorry! I lost track of this PR. I didn't need this feature anymore. But I realized many others do need this. 

So I have updated the branch with the latest `develop` branch, adapted the changes to the new react front-end. Tested locally, works as it was working before. 

I'd appreciate if someone can also test this.

rjblake commented:
> Hey guys, Sorry! I lost track of this PR. I didn't need this feature anymore. But I realized many others do need this.
> 
> So I have updated the branch with the latest `develop` branch, adapted the changes to the new react front-end. Tested locally, works as it was working before.
> 
> I'd appreciate if someone can also test this.

@rumansaleem - happy to test. Can you share more details on how I can go about testing this? Currently installed using docker (image: nginxproxymanager/nginx-proxy-manager-dev:pr-4425)

rumansaleem commented:
@rjblake You can re-pull the same image once the CI build succeeds. Then you can test uploading custom certificates and try updating certificates with/without providing files on the UI.

The build was failing due to outdated invalid swagger schema for the new PUT endpoint. I've fixed it. It should pass this time :crossed_fingers: .

itguy327 commented:
Will this eventually get rolled into the main image?




On Mon, Mar 16, 2026 at 12:56 PM nginxproxymanagerci[bot] <
***@***.***> wrote:

> *nginxproxymanagerci[bot]* left a comment
> (NginxProxyManager/nginx-proxy-manager#4425)
> <https://github.com/NginxProxyManager/nginx-proxy-manager/pull/4425#issuecomment-4069162161>
>
> Docker Image for build 12 is available on DockerHub
> <https://cloud.docker.com/repository/docker/nginxproxymanager/nginx-proxy-manager-dev>
> :
>
> nginxproxymanager/nginx-proxy-manager-dev:pr-4425
>
> Note
>
> Ensure you backup your NPM instance before testing this image! Especially
> if there are database changes.
> This is a different docker image namespace than the official image.
>
> Warning
>
> Changes and additions to DNS Providers require verification by at least 2
> members of the community!
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NginxProxyManager/nginx-proxy-manager/pull/4425#issuecomment-4069162161>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AATOIGA6ZMIO5BJN3NGHN5D4RAW5BAVCNFSM6AAAAABYTNBISOVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHM2DANRZGE3DEMJWGE>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>

rjblake commented:
> @rjblake You can re-pull the same image once the CI build succeeds. Then you can test uploading custom certificates and try updating certificates with/without providing files on the UI.
> 
> The build was failing due to outdated invalid swagger schema for the new PUT endpoint. I've fixed it. It should pass this time 🤞 .

I've re-pulled build, cleared cache and added certificates (through edit). All appears to be working well. Many thanks

rumansaleem commented:
> Will this eventually get rolled into the main image?

@itguy327 I hope so. 

@jc21 can you please have a look?

littlehoushontheprairie commented:
+1

AlyBadawy commented:
+1

newhinton commented:
@rumansaleem 

I know you yourself technically don't need this pr anymore. But i have one small suggestion nonetheless.
For my usecase, i'll need to regularly update the certificates from letsencrypt. I don't want to do that by hand, so i will write myself a script for it if this gets merged.

Now my suggestion:
Can you add some example for an api-call in the ui somewhere?
Even if it is just a barebones curl call, this will be so much better to use for everyone since they don't have to dig through the code or watch their network-traffic.

Thanks!
